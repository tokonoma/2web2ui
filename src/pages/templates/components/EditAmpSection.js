import React from 'react';
import useEditorContext from '../hooks/useEditorContext';
import Editor from './Editor';
import 'brace/mode/html';

const EditAmpSection = () => {
  const { annotations, content, setContent, isReadOnly } = useEditorContext();

  return (
    <Editor
      inlineErrors={annotations.amp_html}
      mode="html"
      name="amp-html-content-editor"
      onChange={value => {
        setContent({ amp_html: value });
      }}
      value={content.amp_html}
      readOnly={isReadOnly}
    />
  );
};

export default EditAmpSection;
