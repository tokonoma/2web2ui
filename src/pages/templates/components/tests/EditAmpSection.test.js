import React from 'react';
import { shallow } from 'enzyme';
import useEditorContext from '../../hooks/useEditorContext';
import EditAmpSection from '../EditAmpSection';

jest.mock('../../hooks/useEditorContext');

describe('EditAmpSection', () => {
  const subject = ({ editorState } = {}) => {
    useEditorContext.mockReturnValue({
      annotations: { amp_html: [{ line: 1, message: 'Oh no!' }] },
      content: { amp_html: '<html ⚡4email>' },
      isReadOnly: false,
      setContent: () => {},
      ...editorState,
    });

    return shallow(<EditAmpSection />);
  };

  it('renders an editor', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('renders correctly in readOnly mode', () => {
    expect(subject({ editorState: { isReadOnly: true } }).prop('readOnly')).toBe(true);
  });

  it('sets content on change', () => {
    const setContent = jest.fn();
    const wrapper = subject({ editorState: { setContent } });

    wrapper.simulate('change', '<html ⚡4email><h1>New Value</h1>');

    expect(setContent).toHaveBeenCalledWith({ amp_html: '<html ⚡4email><h1>New Value</h1>' });
  });
});
