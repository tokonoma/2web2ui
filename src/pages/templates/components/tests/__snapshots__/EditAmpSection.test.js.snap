// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAmpSection renders an editor 1`] = `
<Editor
  inlineErrors={
    Array [
      Object {
        "line": 1,
        "message": "Oh no!",
      },
    ]
  }
  mode="html"
  name="amp-html-content-editor"
  onChange={[Function]}
  readOnly={false}
  value="<html âš¡4email>"
/>
`;
